### PR TITLE
Don't check `vendor/` in to Git

### PR DESCRIPTION
**Pull request by LukeShu**
## Description
I'm sick of this noise when looking at diffs. I already did this in apro.git a while back.

## Related Issues
None.

## Testing
CI worked.

## Checklist

 - [x] I made sure to update `CHANGELOG.md`. - no applicable changes

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [x] This is unlikely to impact how Ambassador performs at scale. - not a runtime change

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [x] My change is adequately tested.

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently. - no tricks

 - [ ] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

### CODE REVIEW

**Review by haq204 — DISMISSED** [review]
omg yes! I never understood why we checked in `vendor/ ` in the first place

**Comment by LanceEa** [comment]
In general, I'm in agreement with this but let's land the rest of the outstanding cycle work first so that we do not end up disrupting cycle work.

**Comment by haq204** [comment]
@LukeShu forgot this after our 3.2.0 and 2.4.1 releases but let's get this landed.
Updating for dependabot stuff is going to get noisy

**Comment by LanceEa** [comment]
@LukeShu @haq204 - I don't see an issue with the proposed changes but I guess why is it that we are keeping the vendor folder around at all? 

I thought the benefit of vendoring was to ensure all users were using the same versions so you don't end up with the "does not work on my machine" situation. The second benefit being if your CI didn't have public internet access, it would not need to grab all these dependencies.

The first one is covered now by Go modules and the second one isn't an issue because Github Actions has access to the public repositories.

**Comment by haq204** [comment]
> @LukeShu @haq204 - I don't see an issue with the proposed changes but I guess why is it that we are keeping the vendor folder around at all?
> 
> I thought the benefit of vendoring was to ensure all users were using the same versions so you don't end up with the "does not work on my machine" situation. The second benefit being if your CI didn't have public internet access, it would not need to grab all these dependencies.
> 
> The first one is covered now by Go modules and the second one isn't an issue because Github Actions has access to the public repositories.

I think the idea is that there isn't? hence why this change to remove the vendor dir. go modules will vendor your deps in GOPATH so I think keeping a vendor/ at all is unnecessary.

File: builder/builder.mk [hunk 1]
```diff
@@ -227,7 +227,7 @@ docker/.base-envoy.docker.stamp: FORCE
 	}
 clobber: docker/base-envoy.docker.clean
 
-docker/.$(LCNAME).docker.stamp: %/.$(LCNAME).docker.stamp: %/base.docker.tag.local %/base-envoy.docker.tag.local %/base-pip.docker.tag.local python/ambassador.version $(BUILDER_HOME)/Dockerfile $(OSS_HOME)/build-aux/py-version.txt $(tools/dsum) FORCE
+docker/.$(LCNAME).docker.stamp: %/.$(LCNAME).docker.stamp: %/base.docker.tag.local %/base-envoy.docker.tag.local %/base-pip.docker.tag.local python/ambassador.version $(BUILDER_HOME)/Dockerfile $(OSS_HOME)/build-aux/py-version.txt $(tools/dsum) vendor FORCE
```
[the remark this answers is not in the document]
Comment (LanceEa) [reply builder/builder.mk:230]: @haq204 @LukeShu - We should get this one landed but just wanted to check-in on this last outstanding item for me...are we just removing it from being checked-in or are we removing the usage of vendoring? Thus no longer needing to run `go mod vendor` when we make changes to deps?

[the remark this answers is not in the document]
Comment (haq204) [reply builder/builder.mk:230]: The only reason I would think we keep `go mod vendor` is if we have deps from private repos that need to be downloaded when building the docker image. But idt that really applies here. So I would favor entirely removing `go mod vendor` from the workflow as well

**Comment by haq204** [comment]
after syncing with @LanceEa and @LukeShu offline, we decided to keep `go mod vendor` for the following reason:
- we want to download the packages outside the docker build to take advantage of caching

It's just that we will not check `/vendor` in git

**Review by LanceEa — APPROVED** [review]

**Review by haq204 — APPROVED** [review]

**Pull request merged**